tools/tests/test_lint.py

```
@@ -5105,3 +5105,277 @@ def test_the_always_on_figure_measures_characters_and_not_bytes(tmp_path):
+    """Criterion 5, second direction.
+
+    Nothing to report is not a failure to derive, and silence would say
+    neither -- so each state produces its own text.
+    """
+    _derivable_tree(tmp_path)
+    note = lint.cell_body_note(tmp_path)
+    assert "no cells" in note, note
+    assert "not derived" not in note, "an empty tree reported a derivation failure"
+
+
+def test_the_charter_row_names_both_budgets_that_bind_it(tmp_path):
```

Ruff (0.16.3)

[warning] Unused function argument: `tmp_path`

(ARG001)